llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -4621,6 +4623,48 @@ static bool isElementRotate(const std::array<std::pair<int, int>, 2> &SrcInfo,
          SrcInfo[1].second - SrcInfo[0].second == (int)NumElts;
 }
 
+static bool isAlternating(const std::array<std::pair<int, int>, 2> &SrcInfo,
+                          const ArrayRef<int> Mask, bool RequiredPolarity) {
```

Comment:
The `const` on ArrayRef isn't doing anything other than making sure you can't change the local variable that contains the start and end pointers. ArrayRef only exposed const* interfaces so you can't change the underlying data. MutableArrayRef is the mutable version.

You can leave `const` if you want. I'm just noting that its uncommon.